content/sensu-core/2.0/reference/agent.md

```diff
@@ -0,0 +1,822 @@
+This makes it possible to generate ad-hoc check requests targeting specific clients via the API.
+
+### Proxy entities
+
+Sensu proxy entities allow Sensu to monitor external resources on systems or devices where a Sensu agent cannot be installed (such a network switch) using the defined [check `proxy_entity_id`][14] to create a proxy entity for the external resource.
+See the [entity reference documentation][3] and the [guide to monitoring external resources][33] for more information about proxy entities.
+
+## Creating events using the agent socket
+
+Every Sensu agent has a TCP, UDP, and HTTP socket listening for external check result input.
+The TCP and UDP sockets listen on the address and port specified by the [socket configuration flags][17]; the HTTP socket listens on the address and port specified by the [API configuration flags][18].
+
+These sockets expect JSON formatted check results, allowing external sources to push check results without needing to know anything about Sensu's internal implementation.
+An excellent agent socket use case example is a web application pushing check results to indicate database connectivity issues.
+
+### Using the TCP socket
+
+The following is an example demonstrating external check result input via the Sensu agent TCP socket.
+The example uses Bash's built-in `/dev/tcp` file to communicate with the Sensu agent socket.
+
+{{< highlight shell >}}
+echo '{"name": "app_01", "output": "could not connect to mysql", "status": 1}' > /dev/tcp/localhost/3030
+{{< /highlight >}}
+
+You can also use [Netcat][19]:
+
+{{< highlight shell >}}
+echo '{"name": "app_01", "output": "could not connect to mysql", "status": 1}' | nc localhost 3030
+{{< /highlight >}}
+
+### Using the HTTP socket
+
+The HTTP socket, just like the TCP and UDP sockets, accepts check results, but it requires a well-formed HTTP request and exposes other functionality that is not possible with the raw TCP/UDP sockets.
+In exchange for a bit more complexity, the HTTP socket interface has the advantage of being more expressive than a TCP/UDP socket, both in the requests that it accepts and how it responds, and so exposes more functionality.
+The following endpoints are available for the HTTP socket:
+
+#### `/healthz` (GET)
+
+This endpoint returns `ok` if the agent is active and connected to a Sensu backend; if the agent is unable to connect to a backend, this endpoint returns `sensu backend unavailable`.
+For example:
+
+{{< highlight shell >}}
+curl -s http://127.0.0.1:3031/healthz
+{{< /highlight >}}
+
+#### `/events` (POST)
+
+This endpoint accepts a JSON [event][7] body and passes the event to the Sensu backend event pipeline for processing. For example:
+
+{{< highlight shell >}}
+curl -s -i \
+-X POST \
+-H 'Content-Type: application/json' \
+-d '{"name": "app_01", "output": "could not connect to mysql", "status": 1}' \
+http://127.0.0.1:3031/events{{< /highlight >}}
```

Comment:
I wasn't able to get this to work based on what I could see in https://github.com/sensu/sensu-go/blob/master/agent/api.go#L53

Reply:
That example worked for the POST API, but I wasn't able to get it to work for the TCP/UDP payload (nightly build). Does this look right?

```
echo '{"check": {"name": "app_01", "output": "could not connect to mysql", "status": 1}}' > /dev/tcp/localhost/3030
```

Reply:
Ha, the payload differs depending on the socket. This is not desirable IMO, I created https://github.com/sensu/sensu-go/issues/2225!

Reply:
Thanks for opening the issue! I've left the two different formats in this doc for now. I'll keep an eye on that issue for changes.